test/unit/ClusterServiceTest.js

```diff
@@ -0,0 +1,404 @@
+chai.use(require('chai-as-promised'));
+const expect = chai.expect;
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+
+const { ClientConfigImpl } = require('../../lib/config/Config');
+const { MemberInfo } = require('../../lib/core/MemberInfo');
+const { ClusterService } = require('../../lib/invocation/ClusterService');
+const { AddressImpl } = require('../../');
+const { UuidUtil } = require('../../lib/util/UuidUtil');
+const { DefaultLogger } = require('../../lib/logging/DefaultLogger');
+const { ConnectionRegistryImpl } = require('../../lib/network/ConnectionRegistry');
+
+describe('ClientClusterServiceImplTest', function () {
+    let loggerStub;
+    let clusterFailoverServiceStub;
+    let connectionRegistryStub;
+    let clientConfig;
+    let clusterService;
+    const INITIAL_MEMBER_LIST_VERSION = -1;
+
+    before(async function () {
+        clientConfig = new ClientConfigImpl();
+    });
+
+    beforeEach(async function () {
+        loggerStub = sandbox.stub(DefaultLogger.prototype);
+        clusterFailoverServiceStub = {
+            current: () => {
+                return {};
+            }
+        };
+        connectionRegistryStub = sandbox.stub(ConnectionRegistryImpl.prototype);
+        clusterService = new ClusterService(
+            clientConfig,
+            loggerStub,
+            clusterFailoverServiceStub
+        );
+        clusterService.translateToAddressProvider.refresh = () => {
+            return Promise.resolve();
+        };
+    });
+
+    afterEach(async function () {
+        sandbox.restore();
+    });
+
+    function createMember({host, port, uuid, liteMember}) {
+        const address = new AddressImpl(host, port);
+        const memberUUID = uuid ? uuid : UuidUtil.generate();
+        return new MemberInfo(address, memberUUID, new Map(), liteMember, null, false, null);
+    }
+
+    it('testMemberAdded', async function () {
+        const members = [];
+        const membershipListener = {
+            memberAdded: (event) => {
+                members.push(event.member);
+            },
+            memberRemoved: () => {}
+        };
+        clusterService.addMembershipListener(membershipListener);
+
+        const member = createMember({host:'127.0.0.1', port:5701, uuid:null, liteMember:false});
+        const clusterUUID = UuidUtil.generate();
+        clusterService.handleMembersViewEvent(connectionRegistryStub, clusterUUID, 1, [member]);
+        const memberInfo = createMember({host:'127.0.0.2', port:5701, uuid:null, liteMember:false});
+        clusterService.handleMembersViewEvent(connectionRegistryStub, clusterUUID, 2, [member, memberInfo]);
+        expect(clusterService.getMembers().length).to.be.equal(2);
+        expect(members).to.eql([memberInfo]);
+    });
+
+    it('testMemberRemoved', async function () {
+        const members = [];
+
+        const memberInfo = createMember({host:'127.0.0.1', port:5701, uuid:null, liteMember:false});
```

Comment:
Can you write the comments in Java ? This comment applies to whole file.

Reply:
Fixed